### PR TITLE
DATAUP-497: Remove the "_batch" message types and use BATCH_ID instead of JOB_ID

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

More simplification of the python side of job messaging. This does the following:
- use `BATCH_ID` instead of `JOB_ID` as the parameter for batch methods
- remove `_batch` from the batch method name
- allow `req.job_id_list` to return `req.job_id` as a list if `req.job_id_list` is not defined
- remove the job request translation (now unnecessary)

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-497
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/biokbase/narrative/jobs/jobcomm.py [hunk 1]
```diff
@@ -55,19 +55,6 @@
     STOP_UPDATE,
 ]
 
-# batch versions
```
Comment (ialarmedalien) [inline src/biokbase/narrative/jobs/jobcomm.py:58]: these are being obliterated

File: src/biokbase/narrative/jobs/jobcomm.py [hunk 2]
```diff
@@ -159,6 +126,8 @@ def job_id(self):
     def job_id_list(self):
         if JOB_ID_LIST in self.rq_data:
             return self.rq_data[JOB_ID_LIST]
+        if JOB_ID in self.rq_data:
+            return [self.rq_data[JOB_ID]]
```
Comment (ialarmedalien) [inline src/biokbase/narrative/jobs/jobcomm.py:130]: allow `req.job_id_list` to return the job_id parameter if it was specified

File: src/biokbase/narrative/jobs/jobcomm.py [hunk 3]
```diff
@@ -471,16 +392,19 @@ def _retry_jobs(self, req: JobRequest) -> dict:
...
+            if "retry" in result
+        ]
+
+        if len(retry_ids):
+            self.send_comm_message(
+                NEW,
+                {
+                    JOB_ID_LIST: retry_ids
+                },
+            )
```
Comment (ialarmedalien) [inline src/biokbase/narrative/jobs/jobcomm.py:407]: only send a list of new IDs if there are actually new IDs!

File: src/biokbase/narrative/tests/test_jobcomm.py [hunk 4]
```diff
@@ -267,20 +262,20 @@ def check_job_id__dne_test(self, request_type):
 
     def check_batch_id__no_job_test(self, request_type):
         for job_id in [None, ""]:
-            req_dict = make_comm_msg(request_type + "_batch", {JOB_ID: job_id}, False)
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/test_jobcomm.py:270]: all these requests are going from being `<request_type>_batch` with param `JOB_ID` to being `<request_type>` with param `BATCH_ID`

File: src/biokbase/narrative/tests/test_jobcomm.py [hunk 5]
```diff
@@ -781,6 +775,23 @@ def test_cancel_jobs__job_id__ok(self):
             request_type=CANCEL, params={JOB_ID: JOB_RUNNING}, ok_states=[JOB_RUNNING]
         )
 
+    def test_cancel_jobs__job_id__dne(self):
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/test_jobcomm.py:778]: a couple of extra tests, just for fun

File: src/biokbase/narrative/tests/test_jobcomm.py [hunk 6]
```diff
@@ -893,6 +904,28 @@ def test_retry_jobs__job_id__ok(self):
             retry_msg,
         )
 
+    @mock.patch(CLIENTS, get_mock_client)
+    def test_retry_jobs__job_id__dne(self):
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/test_jobcomm.py:908]: more fun extra tests!!

File: src/biokbase/narrative/tests/test_jobcomm.py [hunk 7]
```diff
@@ -1434,82 +1461,6 @@ def test_request__no_input(self):
         with self.assertRaisesRegex(ValueError, CELLS_NOT_PROVIDED_ERR):
             req.cell_id_list
 
-    def _check_rq_equal(self, rq0, rq1):
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/test_jobcomm.py:1437]: none of this is needed now as this code was removed

File: src/biokbase/narrative/jobs/jobcomm.py (hunk 1 above)
Comment (briehl) [reply src/biokbase/narrative/jobs/jobcomm.py:58]: 💥

**Review by briehl — APPROVED** [review]
Great! Somehow, I enjoy seeing code get deleted more than getting written.

**Pull request merged**